### ISSUE TITLE
Unit tests: TestUtilitities.isJavaFile is likely to skip package-info.java, maybe other files

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#246]
```
What steps will reproduce the problem?
1. Add this class to unit tests:

in\u0074erface I {
    /*@checkers.quals.Bottom*/public final I i = null;
}

public clas\u0073 C {}

en\u0075m E {}

2. Run the tests.

What is the expected output? What do you see instead?

There should be an error message that matches
//:: error: annotation type not applicable to this kind of declaration

Instead the compilation unit is skipped and no diagnostics are produced.

What version of the product are you using? On what operating system?

Checker Framework 1.6.6, Windows 8.1

Please provide any additional information below.

The file is skipped because isJavaFile method requires that the file content 
contains at least one of the keywords: class, interface or enum. Directory 
scanning method deeplyEnclosedJavaTestFiles() then silently skips those files. 
A workaround is to add one of the keywords to package-info.java file as a 
comment.

```

Original issue reported on code.google.com by `timo.kin...@gmail.com` on 26 Aug 2013 at 1:09
